app/controllers/users_controller.rb

```diff
@@ -0,0 +1,24 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.create(user_params)
+
+    if @user.save
+      flash[:success] = "Thank you for signing up. Please sign in."
+      redirect_to sign_in_path
+    else
```

Comment:
Generally we'd want to test conditional logic in a controller spec. I never test controllers with 'happy path' logic that is exercised in an acceptance test however, and generally speaking when it comes to "routine validation" sad path for CRUD operations I don't write many controller specs for the failure side of conditional logic like that but I'm not opposed to it. In this case it jumps out at me because the alternative is "Something went wrong"... that sounds serious what is it? Validation error? If so we certainly don't want to present a message like this for a validation error. This is part of the reason I'd defer signup because there are more domain specific forms that are more interesting in the near future. If we're gonna do it though, we should do it right and at a minimum this controller logic should be tested but in reality we should have a form builder that presents inline form errors for validation messages which is something I hoped to pair on with you soon regarding the item forms.